### PR TITLE
Update OTel collector deps to v0.102.1

### LINKED ISSUE #26403: [BUG] Opentelemetry package vulnerability
Lately AWS Inspector is alerting about Opentelemetry vulnerability
https://github.com/advisories/GHSA-c74f-6mfw-mm4v

**Agent Environment**

We aren't using the agent directly, but via Lambda extensions

**Describe what happened:**
AWS inspector find a vulnerability in all Lambdas with DD layers

**Describe what you expected:**

**Steps to reproduce the issue:**

**Additional environment details (Operating System, Cloud provider, etc):**
AWS Lambda

### PR DESCRIPTION
**Pull request by dineshg13**
### What does this PR do?
PR updates the dependencies of OTel collector to v0.102.1 and contrib modules to v0.102.0 .  
In the process we had to update OTel go SDK to v1.27.0 .

Collector did a breaking API change for setting configProvider. 
https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.101.0
We need to incorporate the breaking changes in this PR. It is involves using ProviderFactories and ConverterFactories

### Motivation
Update to latest version to pull in the security fixes for a critical CVE.  https://github.com/open-telemetry/opentelemetry-collector/security/advisories/GHSA-c74f-6mfw-mm4v

### Describe how to test/QA your changes
* Run Agent with OTLP enabled and ensure that we are able to send Traces, Metrics and logs via OTLP.

### CODE REVIEW

**Review by mx-psi — COMMENTED** [review]
Looks like we need to address some deprecations on grpc-go

**Review by ajgajg1134 — APPROVED** [review]
looks good for agent-apm

File: comp/otelcol/otlp/collector.go [hunk 1]
```diff
@@ -219,8 +221,18 @@ func NewPipeline(cfg PipelineConfig, s serializer.MetricSerializer, logsAgentCha
 		},
 		BuildInfo:               buildInfo,
 		DisableGracefulShutdown: true,
-		ConfigProvider:          configProvider,
-		LoggingOptions:          options,
+		ConfigProviderSettings: otelcol.ConfigProviderSettings{
+			ResolverSettings: confmap.ResolverSettings{
+				URIs: []string{fmt.Sprintf("map:%s", "hardcoded")},
```
Comment (dustmop) [inline comp/otelcol/otlp/collector.go:226]: Why not?
```suggestion
				URIs: []string{"map:hardcoded"},
```

File: comp/otelcol/otlp/internal/configutils/utils_test.go [hunk 2]
```diff
@@ -53,21 +53,33 @@ func TestNewConfigProviderFromMap(t *testing.T) {
 	require.NoError(t, err)
 	cfgMap, err := NewMapFromYAMLString(string(content))
 	require.NoError(t, err)
-	mapProvider := NewConfigProviderFromMap(cfgMap)
 
+	mapSettings := otelcol.ConfigProviderSettings{
+		ResolverSettings: confmap.ResolverSettings{
+			URIs: []string{fmt.Sprintf("map:%s", "hardcoded")},
```
Comment (dustmop) [inline comp/otelcol/otlp/internal/configutils/utils_test.go:59]: Same question here

File: comp/otelcol/otlp/map_provider_not_serverless_test.go [hunk 3]
```diff
@@ -1110,7 +1113,20 @@ func TestUnmarshal(t *testing.T) {
...
 			},
 		},
-	})
+	}
+	cfgMap, err := buildMap(pcfg)
+	require.NoError(t, err)
+
+	mapSettings := otelcol.ConfigProviderSettings{
+		ResolverSettings: confmap.ResolverSettings{
+			URIs: []string{fmt.Sprintf("map:%s", "hardcoded")},
```
Comment (dustmop) [inline comp/otelcol/otlp/map_provider_not_serverless_test.go:1122]: Same here

**Review by dustmop — COMMENTED** [review]
There a bunch of changes here about `confmap.Converter` and the `ConfigProviderSettings`. Is this a consequence of the dependency version changes, or is it unrelated work? Either way, it would be good to have a description of this change in the PR description.

**Comment by dineshg13** [comment]
@dustmop i have updated PR description with a link to upstream breaking changes.

**Review by dustmop — APPROVED** [review]
LGTM

**Review by jszwedko — APPROVED** [review]

File: go.mod [hunk 4]
```diff
@@ -613,34 +613,34 @@ require (
...
+	github.com/DataDog/datadog-agent/comp/core/status/statusimpl v0.55.0-rc.1
 	github.com/DataDog/datadog-agent/comp/core/telemetry v0.55.0-rc.2
 	github.com/DataDog/datadog-agent/comp/def v0.55.0-rc.2
 	github.com/DataDog/datadog-agent/comp/forwarder/defaultforwarder v0.55.0-rc.2
 	github.com/DataDog/datadog-agent/comp/forwarder/orchestrator/orchestratorinterface v0.55.0-rc.2
 	github.com/DataDog/datadog-agent/comp/logs/agent/config v0.55.0-rc.2
-	github.com/DataDog/datadog-agent/comp/netflow/payload v0.55.0-rc.2
-	github.com/DataDog/datadog-agent/comp/otelcol/collector-contrib/def v0.55.0-rc.2
-	github.com/DataDog/datadog-agent/comp/otelcol/collector-contrib/impl v0.55.0-rc.2
+	github.com/DataDog/datadog-agent/comp/netflow/payload v0.55.0-rc.1
```
Comment (paulcacheux) [inline go.mod:622]: This is probably wrong

**Review by paulcacheux — CHANGES_REQUESTED** [review]

**Review by mx-psi — COMMENTED** [review]
Can you update the PR title to reference v0.102.1 instead?

**Review by mackjmr — APPROVED** [review]

**Review by vickenty — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Review by songy23 — APPROVED** [review]
May be worth adding a changelog with a reference to https://github.com/open-telemetry/opentelemetry-collector/security/advisories/GHSA-c74f-6mfw-mm4v

**Review by DylanLovesCoffee — APPROVED** [review]

**Review by FlorentClarret — APPROVED** [review]

**Review by purple4reina — APPROVED** [review]

**Review by robertjli — APPROVED** [review]

**Review by rafa-jago-scacap — APPROVED** [review]

**Review by CelianR — APPROVED** [review]

**Pull request merged**